integration/querier_remote_read_test.go

```diff
@@ -216,6 +216,7 @@ func TestQuerierStreamingRemoteRead(t *testing.T) {
 
 	httpReq, err := http.NewRequestWithContext(httpReqCtx, "POST", "http://"+querier.HTTPEndpoint()+"/prometheus/api/v1/read", bytes.NewReader(compressed))
 	require.NoError(t, err)
+	httpReq.Header.Add("Accept-Encoding", "snappy")
```

Comment:
Does this trigger the bug (without the fix)? Wondering if this is the test for the bug.

Reply:
Yes. It was explained in the issue: https://github.com/grafana/mimir/issues/2929.